UI/homepage.py

```diff
@@ -85,14 +200,33 @@ def update_items_display():
         delete_buttons.append(btn)
         y_offset += 40  # Adjust y offset for next button
 
+    # Plus Button
+    plus_button = pygame_gui.elements.UIButton(relative_rect=pygame.Rect(370, 100, 30, 30),
+                                               text='+',
+                                               manager=manager,
+                                               object_id=f'plus_button')
+
+    # Value Label
+    value = 0
+    value_label = pygame_gui.elements.UILabel(relative_rect=pygame.Rect(400, 100, 30, 30),
+                                              text=str(value),  # Initial value
+                                              manager=manager,
+                                              object_id=f'value_label')
+
+    # Minus Button
+    minus_button = pygame_gui.elements.UIButton(relative_rect=pygame.Rect(430, 100, 30, 30),
+                                                text='-',
+                                                manager=manager,
+                                                object_id=f'minus_button')
+
 def delete_item(button):
     global item_list
     if button.item_index < len(item_list):
         item_list.pop(button.item_index)
         update_items_display()
 
 def handle_ui_events(event):
-    global current_screen, item_list
+    global current_screen, item_list, value, value_label, check_boxes, recipe_search_button, input_box, item_list, items_text_box, plus_button, minus_button, value_label, filter_title, back_button, search_results
```

Comment:
another instance of potential reinitialization that I mentioned before for the global variables